lib/directory-tree.js

```diff
@@ -27,16 +27,18 @@ function directoryTree (path, options, onEachFile) {
 
 	try { stats = FS.statSync(path); }
 	catch (e) { return null; }
-	if (stats.isFile()) {
-		const ext = PATH.extname(path).toLowerCase();
 
-		// Only add files with the provided extensions
-		if (options && 
-			options.extensions && 
-			options.extensions.length && 
-			options.extensions.indexOf(ext) === -1)
-			return null;
+	// Skip if it matches the exclude regex
+	if (options && options.exclude && options.exclude.test(path))
+		return null;  
 
+	if (stats.isFile()) {
+		// Skip if it does not match the include regex
+		if (options && options.extensions && !options.extensions.test(path))
```

Comment:
Let's only test the extension `ext` below, to be completely predictable and keep things simple.